test/e2e/syncer/syncer_test.go

```diff
@@ -170,6 +170,28 @@ func TestSyncerLifecycle(t *testing.T) {
 	}
 
 	// TODO(marun) Check that the deployment was able to contact kcp
+
+	//Delete the deployment
+	err = downstreamKubeClient.AppsV1().Deployments(downstreamNamespaceName).Delete(ctx, deployment.Name, metav1.DeleteOptions{})
+	require.NoError(t, err)
+
+	//Wait the deployment to be recreated and check it is a different UID
+	t.Logf("Waiting for downstream deployment %s/%s to be created...", downstreamNamespaceName, upstreamDeployment.Name)
+	require.Eventually(t, func() bool {
+		newDeployment, err := downstreamKubeClient.AppsV1().Deployments(downstreamNamespaceName).Get(ctx, upstreamDeployment.Name, metav1.GetOptions{})
+		if apierrors.IsNotFound(err) {
+			return false
+		}
+		if err != nil {
+			t.Errorf("saw an error waiting for downstream deployment %s/%s to be created: %v", downstreamNamespaceName, upstreamDeployment.Name, err)
+		}
+		//Test UID
+		if deployment.UID == newDeployment.UID {
```

Comment:
why this check?

Reply:
In Slack I had suggested that semantic equality + unequal UID would be a rigorous way to check for recreation. IDK if we need it.

Reply:
Make sure it was deleted & recreated